### ISSUE TITLE
[Bug]: BSRN model compatibility issues -- GPU inference exception

### ISSUE THREAD

**Issue by itjianggit** [openvinotoolkit/openvino#19355]
### OpenVINO Version

tag 2023.0.1

### Operating System

Ubuntu 20.04 (LTS)

### Device used for inference

GPU

### Framework

None

### Model used

BSRN

### Issue description

Hi,

I’m trying to use the super-resolution network model bsrn as the openvino inference model.

There is a significant difference in inference performance between CPU and GPU device selection, poor inference performance using GPU. Choosing different models during the training process may result in black images or significant distortion, and the specific reasons are not yet clear - Only for GPU.

Training set: https://data.vision.ee.ethz.ch/cvl/DIV2K/
Paper address: https://arxiv.org/abs/2205.05996
Code address: https://github.com/xiaom233/BSRN

### Step-by-step reproduction

First, download GitHub resources, train to obtain the Pytorch network. 
Second, modify the script to convert to the onnx model. 
![onnx](https://github.com/openvinotoolkit/openvino/assets/69840007/85ad1606-bc23-4cca-8e4e-992cbc52e716)
https://github.com/openvinotoolkit/openvino/assets/69840007/85ad1606-bc23-4cca-8e4e-992cbc52e716

Third, use the mo command to switch to xml + bin. Finally, use the demo (image_processing_demo) of the open_model_zoo to infer.
![mo](https://github.com/openvinotoolkit/openvino/assets/69840007/d75dec8e-a901-449d-bb9f-321f1f753e82)
https://github.com/openvinotoolkit/openvino/assets/69840007/d75dec8e-a901-449d-bb9f-321f1f753e82

./image_processing_demo -m models/model_best.xml -i images/0829x3.png -d CPU -at sr
![cpu](https://github.com/openvinotoolkit/openvino/assets/69840007/f49fe99d-ba38-4c3f-a492-85c52e89d464)
https://github.com/openvinotoolkit/openvino/assets/69840007/f49fe99d-ba38-4c3f-a492-85c52e89d464

./image_processing_demo -m models/model_best.xml -i images/0829x3.png -d GPU -at sr
![gpu](https://github.com/openvinotoolkit/openvino/assets/69840007/cb464298-10f5-428a-80e9-4ca8500d0a96)
https://github.com/openvinotoolkit/openvino/assets/69840007/cb464298-10f5-428a-80e9-4ca8500d0a96

Processed the training dataset and used intel AVS scaling algorithm and jpeg encoding and decoding to obtain small images, the inference effect using GPU is as follows, and the CPU result is normal.
![error](https://github.com/openvinotoolkit/openvino/assets/69840007/8461b318-c189-4b01-ab4e-ac69d298d4fc)
https://github.com/openvinotoolkit/openvino/assets/69840007/8461b318-c189-4b01-ab4e-ac69d298d4fc

### Issue submission checklist

- [X] I report the issue. It's not a question
- [X] I checked the problem with the documentation, FAQ, open issues, Stack Overflow, etc., and have not found the solution
- [X] There is reproducer code and related data files such as images, videos, models, etc.

**Comment by zulkifli-halim** [comment]
Please share with us your ONNX model and the optimized model for replication purposes. To obtain the small images, what command was used?

Also, which GPU did you use for inferencing?

**Comment by itjianggit** [comment]
Hello zulkifi,

I am very sorry that I have been unable to reply to posts on Intel forums. Clicking on the reply button on the official website will only refresh the interface, which has caused me to not reply to your messages in the past few days.

Because I have never known how to contact you, I am unable to provide you with relevant resources. I will send you the model tomorrow.

Finally, I would like to inquire if you know Iffa_Intel. I have also posted on the forum about " GPU zero copy inference exception (remote_blob) " issues, and I am unable to reply to the relevant posts.

Sincerely,

Wanan, Itjianggit

**Comment by itjianggit** [comment]
> Please share with us your ONNX model and the optimized model for replication purposes. To obtain the small images, what command was used?
> 
> Also, which GPU did you use for inferencing?

Hello zulkifi,

I am very sorry that I have been unable to reply to posts on Intel forums. Clicking on the reply button on the official website will only refresh the interface, which has caused me to not reply to your messages in the past few days.

Because I have never known how to contact you, I am unable to provide you with relevant resources. I will send you the model tomorrow.

Finally, I would like to inquire if you know Iffa_Intel. I have also posted on the forum about " GPU zero copy inference exception (remote_blob) " issues, and I am unable to reply to the relevant posts.

Sincerely,

Wanan, Itjianggit

**Comment by zulkifli-halim** [comment]
Hi @itjianggit 

For the "GPU zero copy inference exception (remote_blob)" case. I suggest you open a new thread here on GitHub. @Iffa-Meah will help to assist you.

**Comment by itjianggit** [comment]
> Hi @itjianggit
> 
> For the "GPU zero copy inference exception (remote_blob)" case. I suggest you open a new thread here on GitHub. @Iffa-Meah will help to assist you.

Hello @zulkifli-halim 

Sorry again for replying to you so late. Also. I appreciate the method you provided.

GPU device: 11th Gen Intel(R) Core(TM) i5-1135G7 @ 2.40GHz

The default method for obtaining small images in the DIV2K dataset is bicubic interpolation. (model_best_bicubic.xml)

In order to meet the needs of practical scenarios, we have added Intel's AVS+SFC scaling algorithm, which is provided by Intel. At the same time, we have added the distortion effect of jpeg encoding and decoding, and implemented it using Intel hardware coding to obtain small images using this method. The above methods are all implemented using Intel related code. Using the obtained small images for training

Use the obtained small images for training, and obtain the intermediate model (model_best_45000.xml) and optimal model (with the highest PSNR and SSIM model_best_125000.xml ) for the same validation set.

Due to the normal inference results of the CPU, I suspect that the GPU is not compatible with the model.

Sincerely,
Itjianggit
[models.zip](https://github.com/openvinotoolkit/openvino/files/12449691/models.zip)

**Comment by zulkifli-halim** [comment]
Hi @itjianggit 

I replicated your issue and I received the same result as yours when running inference using GPU on the model_best_125000.xml model:
![image](https://github.com/openvinotoolkit/openvino/assets/79680111/18e4bccc-2389-49a3-8297-4f1e03dbdd92)

The other two models are working fine for both CPU and GPU. I'm looking into this and will update you soon.

**Comment by zulkifli-halim** [comment]
Hi @itjianggit 

It seems that this particular model (model_best_125000.xml) is sensitive to the inferred precision used by the device (i.e. affects the output accuracy). The execution of the GPU defaults to FP16 if not manually configured, which causes output differences when compared to the CPU, which executes as FP32 by default. The solution here is to modify the image_processing_demo (Refer to this 
[main.zip](https://github.com/openvinotoolkit/openvino/files/12481025/main.zip) file) to execute with FP32 precision for GPU, to get the same results as CPU. 
 
Use this command for inference:
_image_processing_demo -i ~/0829x3.png -m ~/models/model_best_125000.xml -d GPU -at sr -infer_precision f32 or f16_

**Comment by itjianggit** [comment]
> Hi @itjianggit
> 
> It seems that this particular model (model_best_125000.xml) is sensitive to the inferred precision used by the device (i.e. affects the output accuracy). The execution of the GPU defaults to FP16 if not manually configured, which causes output differences when compared to the CPU, which executes as FP32 by default. The solution here is to modify the image_processing_demo (Refer to this [main.zip](https://github.com/openvinotoolkit/openvino/files/12481025/main.zip) file) to execute with FP32 precision for GPU, to get the same results as CPU.
> 
> Use this command for inference: _image_processing_demo -i ~/0829x3.png -m ~/models/model_best_125000.xml -d GPU -at sr -infer_precision f32 or f16_

Hi @zulkifli-halim 

Thank you very much for your help. The solution you provided is effective and has successfully solved the problem.

Successfully modified image_ Processing_ The demo is executed with FP32 precision as the GPU, achieving the same results as the CPU.

Sincerely,
Itjianggit

**Comment by zulkifli-halim** [comment]
Hi @itjianggit. Glad that it helps. I'm closing this issue.
